WebApp/src/signaling.ts

```diff
@@ -110,16 +124,48 @@ router.put('', (req: Request, res: Response) => {
 
 router.delete('', (req: Request, res: Response) => {
   const id: string = req.header('session-id');
+  const connectionIds = clients.get(id);
+  if (connectionIds) {
+    connectionIds.forEach(connectionId => {
+      connectionPair.delete(connectionId);
+      offers.delete(connectionId);
+      answers.delete(connectionId);
+    });
+  }
+  candidates.delete(id);
   clients.delete(id);
   res.sendStatus(200);
 });
 
 router.put('/connection', (req: Request, res: Response) => {
   const sessionId: string = req.header('session-id');
-  const connectionId: string = uuid();
+  const { connectionId } = req.body;
+
+  let peerExists = false;
+  if (req.app.get('isPrivate')) {
+    if (connectionPair.has(connectionId)) {
+      const pair = connectionPair.get(connectionId);
+
+      if (pair[0] != null && pair[1] != null) {
+        const err = new Error('this connection is full.');
+        console.log(err);
+        res.status(400).send({ error: err });
+        return;
+      } else if (pair[0] != null) {
+        connectionPair.set(connectionId, [pair[0], sessionId]);
+        peerExists = true;
+      } else if (pair[1] != null) {
```

Comment:
In this program, the process can pass here in a normal scenario?

Reply:
Certainly, process does not pass here.
I will delete